static/panes/tree.ts

```diff
@@ -729,15 +724,6 @@ export class Tree {
         }, newSettings.delayAfterChange);
     }
 
-    private getPaneName() {
-        return `Tree #${this.id}`;
-    }
-
-    private updateTitle() {
```

Comment:
I can see it not being used, but it still confuses me. I guess it gets the container title directly from the factory?

Reply:
no...

Reply:
ah, it IS used, but by some black magic event handling.
![image](https://github.com/compiler-explorer/compiler-explorer/assets/442630/c1ca905e-3567-4bd3-a8b3-0b615ac8ecd7)

So you should not remove these functions, maybe make them public to avoid being removed and then we can see how to make it bit more transparent these things are linked (but I'm not sure how (perhaps best to use an interface, I think the "normal" Pane's do use that))

Reply:
Good catch!  The black magic isn't in event handling, I think it comes down to `PaneRenaming` being completely untyped and in particular uses `pane: any` to successfully call the Pane-private `pane.updateTitle()`, and this is all hidden from TS.  I'll revert the deletion and try to fix it.